libraries/engage/product/components/Swatches/Swatches.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import defaultsDeep from 'lodash/defaultsDeep';
+import { css } from 'glamor';
+import { PRODUCT_SWATCHES } from '@shopgate/pwa-common-commerce/product/constants/Portals';
+import { SurroundPortals } from '../../../components';
+import { isBeta, useConfig, useWidgetConfig } from '../../../core';
+import { Swatch } from '../Swatch';
+import connect from './connector';
+
+const { typography = {} } = useConfig();
+
+const widgetDefaults = {
+  settings: {
+    // Filters out all swatches which are not in the list, no filter = [] => show all
+    maxItemCount: 10,
+    filter: [],
+  },
+  styles: {
+    swatches: {
+      lineHeight: typography.lineHeight,
+    },
+  },
+};
+
+const widgetId = '@shopgate/engage/product/Swatches';
+
+/**
+ * Renders only product swatches from a list of characteristics.
+ * @param {Object} props The component props.
+ * @returns {JSX}
+ */
+const SwatchesUnwrapped = ({ productId, characteristics, widgetPath }) => {
```

Comment:
We do not name component in this manner. Do we have benefits of that? `displayName` is designed for debugging purpose only.

Reply:
Not only debugging but also maintenance, which this is actually for.